src/app/chat/views/channels_list.nim

```diff
@@ -112,3 +112,10 @@ QtObject:
     let topLeft = self.createIndex(idx, 0, nil)
     let bottomRight = self.createIndex(idx, 0, nil)
     self.dataChanged(topLeft, bottomRight, @[ChannelsRoles.Timestamp.int, ChannelsRoles.LastMessage.int, ChannelsRoles.UnreadMessages.int])
+
+  proc clearUnreadMessagesCount*(self: ChannelsList, name: string) =
```

Comment:
Not very familiar with the code, so forgive me if it's already done.

To clear messages count it's not enough to set the count to `0` in the chat struct and save (we you might still want to do, as unviewed messages count is managed by the client), because messages are still marked individually as not read, and sometimes we re-calculate the denormalized message count from the messages collection.

What you want to do is either call `MarkMessagesSeen` on the individual message with an array of `messageIds` https://github.com/status-im/status-go/blob/ee0a83fdc4a19f70fa8d08347ef67c0ca3d67f1e/protocol/messenger.go#L2112 , or you can call `MarkAllRead` with the given chatID https://github.com/status-im/status-go/blob/ee0a83fdc4a19f70fa8d08347ef67c0ca3d67f1e/protocol/messenger.go#L2128.

Reply:
Ah, thanks for pointing that out! I'll take a look!